### DISCUSSION TITLE
Clarification Rate limited excceded Error

### DISCUSSION THREAD

**Discussion by Hunt092** [microsoft/vscode-discussions#1850] [Extension Development QnA]
while developing an extension after a few calls i am randomly getting an rate limited exceeded error, in the documentation i see no clarification as to what the limit is, i am making sure that we don't cross the token limit per request, but is there a limit of how many request can be done by each account, is there any difference of the rate for business tier and enterprise tier.

How does the quota work, how will the request limit replenish, does the rate counter started with the first request? or is a time based (like x per hour )

**Comment by alefragnani** [comment]
Hi @Hunt092 ,

Could you provide more details about which service are you calling? Is it GitHub, ChatGPT or something else?

Thank you

**Reply by Hunt092** [reply]
hi @alefragnani ,

i am calling the vscode.lm langauge model with gpt4o as the model family

**Reply by alefragnani** [reply]
Based on the documentation (https://code.visualstudio.com/api/extension-guides/language-model#rate-limiting ), there is indeed a rate limit, and VS Code is transparent about it.

I’ve found at least two closed issues (when was still on _proposed_ state) about rate limit on vscode lm, that is probably similar to the issue you are facing. 

* https://github.com/microsoft/vscode/issues/204266
* https://github.com/microsoft/vscode/issues/201996

Let me know if you need more details

Hope this helps

**Reply by Hunt092** [reply]
They both refer to different rate limits,
one https://github.com/microsoft/vscode/issues/204266 refers to per request max token limit 
second https://github.com/microsoft/vscode/issues/201996 refer to github login limit

what i am refering/ facing issue with is the copilot llm request limit ( i am ensuring per request token limit is not reached ) but i am getting rate limited on the amount of request which are happening, and i just want to understand the how that rate limiting works, what is the limit and things around the same

**Reply by Hunt092** [reply]
Hi @alefragnani,

let me rephrase it to make it much clear, 
the Rate limit of copilot LLM discussion is talking about the context window limit of we can send only x max token (i think the number is around 20k fot gpt4o) per llm call (request)

The limit me and my team are hitting up are the over how many llm call can be made, because after calling the LLM say 10 - 20 or even sometimes 50th times is fine, but the 51st call gives me tate limited exceeded

**Reply by alefragnani** [reply]
Hi @Hunt092 ,

I think, for this specific details, we will need help from someone from the VS Code team. @olguzzar , could you redirect/suggest someone to help us on this?

Thank you

**Reply by Hunt092** [reply]
Hey @olguzzar, 

Can you help out with this issue?

**Reply by olguzzar** [reply]
tagging in @isidorn

**Reply by isidorn** [reply]
Thanks for opening this discussion.
I am curious what kind of extension are you building?

As for the tokens limit - we are constantly increasing those, and I think the new token limit size will be 64k.

As for the limit for the number of requests, we have a limit but let me find our exactly how large this limit is. Are you hitting this all the time? What are your reproducible steps?

**Reply by Hunt092** [reply]
Hi @isidorn,

Just as an overview of what we are doing, my team and i are building a transformation tool suite with Custom Chat participant so we go over multiple type so files, and have llm help us do some summarization / transformation, typically per file we have like 3x or 4x call, so once the process starts we will be making alot of calls to the extending of few hundreds for bigger chunks in a single command.

i am ensuring the token limit is not hit ( though having those extra tokens will be a blessing), but i am facing the rate limit issue when going for those bigger chunks of files, 

i am also generally curious over how this rate limit works, is it time dependent, or there is a limit to parallel calls, is there a difference for enterprise vs business user for this limits

**Reply by isidorn** [reply]
There is not difference in limit for business vs enterprise that I know.
@thispaul can maybe point you to some documentation on GH Copilot limits.

**Reply by Hunt092** [reply]
HI @thispaul, can you provide any updates, or you if you need anymore information

**Reply by Hunt092** [reply]
@isidorn or @alefragnani , any help you guys can provide?

**Reply by burkeholland** [reply]
Pinged @thispaul on this

**Comment by theReynald** [comment]
@thispaul, Any insight on this question?

**Reply by Hunt092** [reply]
hey @theReynald , do you think there is  anyone else  who can help apart from Paul?
